saber/sequence_processor.py

```diff
@@ -177,34 +175,37 @@ def save(self, filepath, compress=True, model=0):
         return True
 
     def load(self, filepath):
-        """Coordinates the saving of Saber models.
+        """Coordinates the loading of Saber models.
 
-        Loads the necessary files for model creation from filepath.
+        Loads the necessary files for model creation from filepath. Creates and compiles a
+        Keras model based on these files.
 
         Args:
-            filepath (str): directory path to saved pretrained folder
+            filepath (str): directory path to saved pre-trained model folder
         """
-        #
         generic_utils.decompress_model(filepath)
+
         # create filepaths
         weights_filepath = os.path.join(filepath, 'model_weights.hdf5')
         attributes_filepath = os.path.join(filepath, 'model_attributes.pickle')
 
-        # load attributes
+        # load attributes, these attributes must be carried over from saved model
         model_attributes = pickle.load(open(attributes_filepath, "rb"))
-        # these attributes must be carrie over from pre-trained model
-        self.config.word_embed_dim = model_attributes['config'].word_embed_dim
-        self.config.char_embed_dim = model_attributes['config'].char_embed_dim
+        self.config.word_embed_dim = model_attributes['word_embedding_dim']
+        self.config.char_embed_dim = model_attributes['char_embedding_dim']
+        self.token_embedding_matrix = model_attributes['embedding_matrix']
 
-        self.ds = model_attributes['ds']
-        self.token_embedding_matrix = model_attributes['embeddings']
+        # create a new dataset instance
+        # TEMP: this is an ugly hack, need way around having to provide a filepath
+        dummy_ds = os.path.abspath('saber/tests/resources/dummy_dataset_1')
```

Comment:
This can't be the best solution to the problem. The dataset object _requires_ a valid dataset passed as the argument `filepath` at construction, so my ugly fix is to use the same dummy dataset used for the unit tests.